README.md

```diff
@@ -1,56 +1,72 @@
+### websocket 
+This package is to handle interactions with the websocket. It has a hub, which handles the publish-subscribe paradigm, and
+the hub has an actor (either Organizer or Witness), that implements the PoP concept.
 
-## Dependencies
+### define
+This package provides structure for "real world events" and corresponding jsonRPC messages,  methods to convert byte received arrays into theses structures
+and function to check validity of the messages and signatures we received.
+
+### jsonRPC
+This package is not used at all, but is defines the protocol we use to format the messages we send and receive.
+
+### test
+This package is, as its name says it, for the tests.
+
+## Back-end components
+
+### Dependencies
 Currently, the project works with the following libraries:
 * [gorilla/websocket](https://github.com/gorilla/websocket): a websocket package for golang. We chose to use this 
 package because it offers a good API, and it is more complete than the websocket package offered by the standard library.
 * [boltdb](https://github.com/boltdb/bolt): a low-level key-value DB management system. It works by creating Buckets, 
 that we can populate with key-value pairs.
 
-## WebSocket Protocol
-The actual implementation (4 files) currently works as following :
-* the `hub.go` maintains a list of opened connections and "routes" the messages. We will change this so there is no more routing to do in the next update.
-* the `connection.go` is the file that properly handles the websocket connection. When it receives a message from the client, it passes 
-it to the hub.
-* the `index.html` acts like the client that prints messages received from the server and that can send messages to the server.
-* the `homePage.go` just serves the `index.html` file on http requests
-### tweaking the server's properties 
+### network testing 
 If you want to test the websocket protocol the server running on a different machine than the client, you have to set the
 server's IP address in `index.html`  at line 36 (instead of `localhost`). You'll have to put your computer's IP address
 before the ":8080" of `main.go` at line 29.
 
-## Database Structure
-As we are using a very basic DBMS enabling only key-values storage. There are currently 3 different databases :
-1. for the hub, containing a list of channels and their subscribers
-2. one for the witness (described below)
-3. one for the organizer (same structure as witness's one)
+### Database Structure
+Currently, there are 2 databases created/opened when we run the backend: orgDatabase.db/witDatabase.db and sub.db.
+
+#### Sub.db
+This database exists only to handle publisher and subscribers of channels. Currently, we only manage subscribers. There is a bucket called
+"sub", in which we store the (key, value) pairs as following : each key is the ID of one channel, and the value is an array of connections
+that subscribed to this channel. When someone subscribes or unsubscribes, we simply edit this array.
```

Comment:
does this have to be persisted to disk? the connections are lost if someone disconnects, right?

Reply:
Well, you're right. I did not think of that as I first implemented this believing we would know the connection's public keys, and then I just tweaked my implementation to make it work without the Pkeys. I'll edit this soon.

Reply:
Well I think I'm going to implement this for the next PR, as it will then make sense to have the hub managing the Publish-Subscribe level. But this would imply that the Hub make a first level of Unmarshalling the message. This is not a problem, but as the code works in it's current state, I prefer to postpone this.